### DISCUSSION TITLE
wecom Add a new alert channel

### DISCUSSION THREAD

**Discussion by aimuz** [grafana/grafana#55883] [Alerting]
The current wecom only supports the use of group bot channels, I want to extend it and add new channels, wecom currently supports two channels

- group bot
-  api

I want to add an api-based channel that can be used for more scenarios.

**Comment by joeblubaugh** [comment]
I think we could do this without creating a new notifier type by extending the existing WeCom notifier. As an example, Slack provides configuration to use either a webhook or a set of credentials and a channel name to send messages.

We could do something similar with WeCom, providing config for both a group bot or an API call.

**Reply by aimuz** [reply]
Yes, that's the way I'm doing it in my currently uncommitted code, and I'm waiting for https://github.com/grafana/grafana/pull/55337

**Reply by joeblubaugh** [reply]
Got it. So the proposal is "Modify the existing wecom notifier."

If you don't want to wait for ShowWhen to be fixed in the frontend you can use the approach that the Slack notifier takes of validating optional fields and returning an error.

**Reply by aimuz** [reply]
@joeblubaugh You can see that I created a PR https://github.com/grafana/grafana/pull/55991

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
